xarray/core/dataset.py

```diff
@@ -1449,6 +1449,14 @@ def __len__(self) -> int:
         return len(self.data_vars)
 
     def __bool__(self) -> bool:
+        warnings.warn(
+            "coercing a Dataset to a bool will be deprecated. "
+            "Using bool(ds.data_vars) to check for at least one "
+            "data variable or using Dataset.to_array to test "
+            "whether array values are true is encouraged.",
+            PendingDeprecationWarning,
+            stacklevel=2,
```

Comment:
I copied this from elsewhere but don't know when to use various stacklevels. this could use some eyeballs.

Reply:
reading more about this I think maybe the `stacklevel` argument can be dropped (`stacklevel=1`) since Dataset.__bool__ isn't wrapping anything - that seem right?

Reply:
ok I checked this out and the current implementation seems fine. stacklevel=2 will mean a warning will be raised on `bool(ds)` rather than within `ds.__bool__` which seems right.